blocks/rich-text/index.js

```diff
@@ -441,6 +460,60 @@ export class RichText extends Component {
 		};
 	}
 
+	getInsertionPosition() {
+		const container = this.getContainerNode();
+		const marginLeft = window.getComputedStyle( container )[ 'margin-left' ];
+		const containerPosition = container.getBoundingClientRect();
+		const rect = getRectangleFromRange( this.editor.selection.getRng() );
+
+		return {
+			top: rect.top - containerPosition.top,
+			left: rect.right - containerPosition.left,
+			marginLeft,
+			height: rect.height,
+		};
+	}
+
+	insertInlineBlock() {
+		const { inlineBlock, completeInlineInsert } = this.props;
+
+		if ( inlineBlock.type === 'image' ) {
+			const { url, alt, width } = inlineBlock;
+			let img;
+
+			if ( width > 150 ) {
+				img = `<img src="${ url }" alt="${ alt }" width="150" />`;
```

Comment:
Shouldn't this be defined somewhere else (by the block/token)?

Reply:
Agreed. There has been some discussion on building an API for Inline Blocks in https://github.com/WordPress/gutenberg/pull/5794#issuecomment-377117224 and https://github.com/WordPress/gutenberg/pull/5794#issuecomment-377912079 but I haven't gotten to that yet. Would be great to get some thoughts on how best to approach that.

Reply:
https://github.com/WordPress/gutenberg/pull/5794/commits/06ce12556d7bfae783db4a857edca002ec1b91e5 adds a module for inline blocks and moves this logic there.